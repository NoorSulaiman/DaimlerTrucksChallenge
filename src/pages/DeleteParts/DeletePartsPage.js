import React, { useState, useContext } from "react";
import { appContext } from "../../provider";
import GroupsList from "../../components/GroupsList/GroupsList";
import ComponentCardList from "../../components/ComponentsList/ComponentCardList";

export default function PartSortingPage() {
  const context = useContext(appContext);
  const groupsArray = context.parts;
  const [componentsList, setComponentsList] = useState([]);

  const sortComponents = (group) => {
    let sortedComponentsArray = [];
    groupsArray.map((component) => {
      let prefix = component.location.slice(0, 2);
      if (group === prefix) {
        sortedComponentsArray.push(component);
      }
    });

    sortedComponentsArray = sortedComponentsArray.sort((a, b) =>
      a.location > b.location ? 1 : b.location > a.location ? -1 : 0
    );
    setComponentsList(sortedComponentsArray);
  };
  const deleteComponent = (locationName) => {
    let newComponentsArray = componentsList;
    context.parts.map((component, indx) => {
      if (locationName === component.location) {
        // Removed item will not be shown in Part Sorting page, as it's been removed from the context
        context.parts.splice(indx, 1);
        newComponentsArray.splice(indx, 1);
        setComponentsList(newComponentsArray);
      }
    });
  };
  return (
    <div>
      {groupsArray.length && (
        <GroupsList components={groupsArray} selectGroup={sortComponents} />
      )}
      {componentsList.length ? (
        <ComponentCardList
          components={componentsList}
          cardButtonFunction={deleteComponent}
        />
      ) : null}
    </div>
  );
}
