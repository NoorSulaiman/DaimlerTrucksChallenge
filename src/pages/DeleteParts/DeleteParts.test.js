import React from "react";
import { shallow } from "enzyme";
import DeletePartsPage from "./DeletePartsPage";

const compArray = [
  {
    panelNumber: "22-58883-000",
    panelDescription: "PNL-INR TRIMPLTKAHLUA SPRKL",
    location: "AGAAA",
    parentPanelNumber: "734-C02202",
    modelCode: "H",
    whitelist: ["AGAAA"],
    blacklist: [],
    type: "hardwired-component",
    switchId: "",
    bom: "734-C02202",
    isAsBuilt: true,
  },
  {
    panelNumber: "22-69410-000",
    panelDescription: "PNL-SW,SMARTPLX,OVHD,SD,KAHLUA",
    location: "OVLHP",
    parentPanelNumber: "679-C01232",
    modelCode: "H",
    whitelist: ["OVLHP", "OVRHP"],
    blacklist: [],
    type: "hardwired-component",
    switchId: "",
    bom: "679-C01232",
    isAsBuilt: true,
  },
  {
    panelNumber: "A06-30769-027",
    panelDescription: "SW-RCKR,PS PWR WDW",
    location: "AG017",
    parentPanelNumber: "66A-C00132",
    modelCode: "H",
    whitelist: ["AG003"],
    blacklist: [],
    type: "hardwired-component",
    switchId: "NONE",
    bom: "66A-C00132",
    isAsBuilt: true,
  },
];
const deleteComponent = (locationName) => {
  let newComponentsArray = compArray;
  compArray.map((component, indx) => {
    if (locationName === component.location) {
      newComponentsArray.splice(indx, 1);
    }
  });
  return newComponentsArray;
};
describe("DeletePartsPage", () => {
  it("should render PartSorting Page correctly with given props", () => {
    const component = shallow(<DeletePartsPage />);
    expect(component).toMatchSnapshot();
  });

  it("should sort components list into group prefixes", () => {
    const component = shallow(<DeletePartsPage />);
    expect(component.find("GroupsList").length).toBe(1);
  });

  it("should take prefixs and sort them", () => {
    expect(deleteComponent("AGAAA").length).toStrictEqual(2);
  });
});
