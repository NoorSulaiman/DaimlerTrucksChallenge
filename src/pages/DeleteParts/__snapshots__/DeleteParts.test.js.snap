// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeletePartsPage should render PartSorting Page correctly with given props 1`] = `ShallowWrapper {}`;
